### PR TITLE
Fix moreh_clip_grad_norm

### PR DESCRIPTION
**Pull request by DuongQLee**
### Ticket
Link to Github Issue: https://github.com/tenstorrent/tt-metal/issues/15690

### Problem description
moreh_clip_grad_norm currently contains bug in implementation.

### What's changed
I fixed the bugs, mainly in program_factory for step 3 of moreh_clip_grad_norm, remove deprecated API in kernel code and refactor test accordingly to current standard. moreh_clip_grad_norm test now pass with atol=rtol=0.1

### Checklist
- [x] Post commit CI passes: https://github.com/tenstorrent/tt-metal/actions/runs/12273241936
- [x] Blackhole Post commit (if applicable): NA
- [x] Model regression CI testing passes (if applicable): NA
- [x] Device performance regression CI testing passes (if applicable): NA
- [x] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes: NA
- [x] New/Existing tests provide coverage for changes: 18/18 test passed

### CODE REVIEW

**Comment by razorback3** [comment]
Also targets https://github.com/tenstorrent/tt-metal/issues/15390

**Review by seunghwan100 — APPROVED** [review]

**Review by razorback3 — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py [hunk 1]
```diff
@@ -88,66 +67,70 @@ def test_moreh_clip_grad_norm(
             param.grad = grad
 
             cpu_inputs.append(param)
```
Comment (thanhnguyen-moreh) [inline tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py:69]: Can refactor to torch/ttnn but doesn't matter too much

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ Tensor MorehClipGradNorm::invoke(
     const auto num_iter = (total_num_inputs + max_num_inputs - 1) / max_num_inputs;
 
     auto tmp_pow_sum = create_device_tensor(
-        SimpleShape{tt::constants::TILE_HEIGHT, tt::constants::TILE_WIDTH * static_cast<uint32_t>(inputs.size())},
+        SimpleShape{static_cast<uint32_t>(inputs.size()), 1, 1},
```
Comment (thanhnguyen-moreh) [inline ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm.cpp:46]: It's better if we know what is the meaning of each component

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/moreh_clip_grad_norm_step1_kernel.cpp [hunk 3]
```diff
@@ -58,9 +58,8 @@ void MAIN {
     for (uint32_t tile_idx = 0; tile_idx < num_tiles; tile_idx++) {
         // Comput cb_xabs and mask(optional)
         // |x|
-        ACQ();
+        tile_regs_acquire();
         cb_wait_front(cb_x, onetile);  // comes from the reader
-        cb_reserve_back(cb_xabs, onetile);
```
Comment (thanhnguyen-moreh) [inline ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/moreh_clip_grad_norm_step1_kernel.cpp:63]: You can keep reserve_back/push_back and wait_front/pop_front at the start/end of each computation step to keep track better of which cb you're reading/writing.

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/moreh_clip_grad_norm_step2_kernel.cpp [hunk 4]
```diff
@@ -40,38 +40,37 @@ void MAIN {
     // Compute cb_x
     for (uint32_t tile_idx = 0; tile_idx < num_tiles; tile_idx++) {
         if (tile_idx == 0) {
-            ACQ();
+            tile_regs_acquire();
             cb_wait_front(cb_input, onetile);  // comes from the reader
-            cb_reserve_back(cb_x, onetile);
```
Comment (thanhnguyen-moreh) [inline ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/moreh_clip_grad_norm_step2_kernel.cpp:45]: Same thing above

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_device_operation.cpp [hunk 5]
```diff
@@ -37,7 +37,7 @@ void MorehClipGradNormStep2Operation::validate_on_program_cache_hit(
 
 MorehClipGradNormStep2Operation::shape_return_value_t MorehClipGradNormStep2Operation::compute_output_shapes(
     const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
-    return SimpleShape{tt::constants::TILE_HEIGHT, tt::constants::TILE_WIDTH};
+    return SimpleShape{1, 1};
```
Comment (thanhnguyen-moreh) [inline ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_device_operation.cpp:40]: Kind of magic numbers, please comment out their meaning if possible

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step3/device/kernels/moreh_clip_grad_norm_step3_kernel.cpp [hunk 6]
```diff
@@ -25,18 +25,18 @@ void MAIN {
 
     // Compute cb_y
     for (uint32_t tile_idx = 0; tile_idx < num_tiles; tile_idx++) {
-        ACQ();
+        tile_regs_acquire();
         cb_wait_front(cb_x, onetile);  // comes from the reader
-        cb_reserve_back(cb_y, onetile);
```
Comment (thanhnguyen-moreh) [inline ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step3/device/kernels/moreh_clip_grad_norm_step3_kernel.cpp:30]: Same thing above

**Review by thanhnguyen1512 — CHANGES_REQUESTED** [review]

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_device_operation.cpp (hunk 5 above)
Comment (DuongQLee) [reply ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_device_operation.cpp:41]: resolved in [ce8237a](https://github.com/tenstorrent/tt-metal/pull/15897/commits/ce8237a7c45fd186600377fcd4f57975e5e5fd4e)

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step3/device/kernels/moreh_clip_grad_norm_step3_kernel.cpp (hunk 6 above)
Comment (DuongQLee) [reply ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step3/device/kernels/moreh_clip_grad_norm_step3_kernel.cpp:30]: resolved in [ce8237a](https://github.com/tenstorrent/tt-metal/pull/15897/commits/ce8237a7c45fd186600377fcd4f57975e5e5fd4e)

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/moreh_clip_grad_norm_step2_kernel.cpp (hunk 4 above)
Comment (DuongQLee) [reply ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/kernels/moreh_clip_grad_norm_step2_kernel.cpp:45]: resolved in [ce8237a](https://github.com/tenstorrent/tt-metal/pull/15897/commits/ce8237a7c45fd186600377fcd4f57975e5e5fd4e)

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/moreh_clip_grad_norm_step1_kernel.cpp (hunk 3 above)
Comment (DuongQLee) [reply ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/moreh_clip_grad_norm_step1_kernel.cpp:63]: resolved in [ce8237a](https://github.com/tenstorrent/tt-metal/pull/15897/commits/ce8237a7c45fd186600377fcd4f57975e5e5fd4e)

File: ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm.cpp (hunk 2 above)
Comment (DuongQLee) [reply ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm.cpp:46]: resolved in [ce8237a](https://github.com/tenstorrent/tt-metal/pull/15897/commits/ce8237a7c45fd186600377fcd4f57975e5e5fd4e)

File: tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py (hunk 1 above)
Comment (DuongQLee) [reply tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py:69]: I think this works well enough.

**Review by thanhnguyen1512 — APPROVED** [review]

**Pull request merged**